cpp/src/copying/slice.cu

```diff
@@ -18,7 +18,7 @@
 #include <cudf/copying.hpp>
 #include <cudf/detail/copy.hpp>
 #include <cudf/detail/iterator.cuh>
-#include <cudf/detail/null_mask.hpp>
+#include <cudf/detail/null_mask.cuh>
```

Comment:
This doesn't look correct to me.  I am not seeing any usage of the functions inside `null_mask.cuh` in `slice.cu`.

I believe this also means you can keep this as a `.cpp` file.